W-16778879 HL7 Connector 4.3.0 Release (#2698)
mulesoft/docs-connectors: hl7/4.3/modules/ROOT/pages/index.adoc

Here is an example:

[source,xml,linenums]
----
    <hl7:config name="HL7_Config" doc:name="HL7 EDI Config" eventMessageMap="/hl7/v2_3/event-message.yaml">
		<hl7:schemas>
			<hl7:schema value='/hl7/v2_3/ADT_A01.esl' />
			<hl7:schema value="/hl7/v2_3/ACK.esl" />
		</hl7:schemas>
	</hl7:config>
    <hl7:read config-ref="HL7_Config" doc:name="HL7 EDI"/>
    ...
    <ee:transform-message doc:name="Create Outgoing Message">
        <ee:set-payload><![CDATA[%dw 2.0
            output application/java
            ---
            {
              Name: "ACK",
              MSH: payload.ACK.MSH,
              Id: "ACK",
              Data: {
                ACK: payload.ACK
              }
            }]]>
        </ee:set-payload>
    </ee:transform-message>
    <hl7:write config-ref="HL7_Config" doc:name="ACK"/>
    ...
    <file:write doc:name="Write" path="output/ack.edi"/>
----

The generated ACK messages have MSH data set up for sending back to the sender of the original message, so you don't need to change anything in the data to perform the send.

> If you include an ACK message schema in your configuration, that schema is used to both receive and generate ACK messages, you also need to specify include the `eventMessageMap` to use the ACK structure in the parser. If you don't specify an ACK schema, the standard `hl7/v2_5_1/ACK.esl` schema default is used.

Reviewer: ```suggestion
If you include an ACK message schema in your configuration and that schema is used to receive and generate ACK messages, you must specify the `eventMessageMap` value to enable the ACK structure in the parser. If you don't specify an ACK schema, the configuration uses the standard `hl7/v2_5_1/ACK.esl` schema by default.
```
